jscomp/core/lam.ml

```diff
@@ -532,7 +543,8 @@ let prim ~primitive:(prim : Lam_primitive.t) ~args loc : t =
           Lift.string (a ^ b)
       | ( (Pstringrefs | Pstringrefu),
           Const_string { s = a; unicode = false },
-          Const_int { i = b } ) -> (
+          Const_int { i = b; comment = pointer_info } ) -> (
+          let b = extract_const_as pointer_info b in 
```

Comment:
At this point, after seeing all these case, why not having this directly inside `Const_int`?
Or rather, having this done, in whatever form, at `Const_int` creation (or earlier)?

Reply:
Another way of asking the question: are there cases where the "original" int is what's needed, and cases where the override one is what's needed.
If both are needed, 2 values need to be accessed/stored.
Otherwise, it's cleaner to ever only have 1.

Reply:
After further checking, there are only several cases that match `Lconst (Const_int ...)` like in the following code,

 https://github.com/shulhi/rescript-compiler/blob/e6fb6e54e2f949b9b08a4d8da8cb1c0f32fd948c/compiler/core/lam_compile.ml#L1774-L1783

I've fallback to the original implementation since it is simpler but it does change the representation of variant in runtime (just for `int` case). I don't think it affect anything else and I've added several more coercion tests just in case to cover cases like in the above linked code.

> Another way of asking the question: are there cases where the "original" int is what's needed, and cases where the override one is what's needed.

As far as I understand, the override one is still needed for case where the `as` payload is not an `int`. The "original" int is needed for runtime representation for the variants.

Reply:
@shulhi I don't follow entirely, how is the runtime representation changed?